tools/extra/fpgadiag/nlb0.cpp

```diff
@@ -310,14 +314,57 @@ bool nlb0::setup()
     options_.get_value<bool>("suppress-hdr", suppress_header_);
     options_.get_value<bool>("csv", csv_format_);
 
+    options_.get_value<std::string>("id", nlb0_id_);
+    fpga_guid nlb0_id;
+
+    if (uuid_parse(nlb0_id_.c_str(), nlb0_id)<0) {
+        std::cerr << "Invalid NLB0 id:" << nlb0_id_ << std::endl;
+        return false;
+    }
+
+    uint32_t offset = 0;
+    uint64_t dfh = 0;
+
+    uint8_t* n = static_cast<uint8_t *>(&nlb0_id[0]);
+    uint64_t nlb0_hi = bswap_64(*n);
+    uint64_t nlb0_lo = bswap_64(*(n+8));
+
+    // enumerate dfh if input id argument 
+    if (options_.find("id")) {
+
+        bool found = false;
+
+        do {
+             uint64_t feature_uuid_lo, feature_uuid_hi;
+             // Read the next feature header
+             dfh = accelerator_->read_csr64(static_cast<uint32_t>(offset));
+             feature_uuid_lo = accelerator_->read_csr64(static_cast<uint32_t>(offset + 8));
+             feature_uuid_hi = accelerator_->read_csr64(static_cast<uint32_t>(offset + 16));
+             if ((nlb0_lo == feature_uuid_lo) && (nlb0_hi == feature_uuid_hi)) {
+                  offset_ = offset;
+                  found = true;
+                  printf("found the NLB offset=0x%x\n", offset);
+                  break;
+             }
+
+            offset += NEXT_DFH_OFFSET(dfh);
+        } while (!DFH_EOL(dfh));
+
+        if (!found) {
+            std::cerr << "Not found NLB:" << nlb0_id_ << std::endl;
+            return false;
```

Comment:
This check here seems to prevent nlb from running on vc as they have different afu guids.